### PR TITLE
C#: Add implementation for lowering and lifiting flags

### PR DESCRIPTION
**Pull request by yowl**
.. and enable the corresponding codegen test.  Implemented in c# as `enum`s.

(contributes to making the `records` runtime test pass)

cc @silesmo @jsturtevant

### CODE REVIEW

File: crates/csharp/src/lib.rs [hunk 1]
```diff
@@ -1175,24 +1175,22 @@ impl<'a> wit_bindgen_core::InterfaceGenerator<'a> for InterfaceGenerator<'a> {
...
                 };
-                format!(
-                    "public static readonly {name} {flag_name} = new {name}(({ty}) (1{suffix} << {i}));"
-                )
+                format!("{flag_name} = 1{suffix} << {i},")
             })
             .collect::<Vec<_>>()
             .join("\n");
 
+        let enum_type = if matches!(flags.repr(), FlagsRepr::U32(2)) {
```
Comment (silesmo) [inline crates/csharp/src/lib.rs:1183]: Shouldn't this be set to the corresponding type based on the number of flags as in the specification as seen here? https://github.com/WebAssembly/component-model/blob/main/design/mvp/CanonicalABI.md#alignment

File: tests/README.md [hunk 2]
```diff
@@ -25,6 +25,12 @@ To run just `codegen` tests for a single language (replace rust with language of
 cargo test -p wit-bindgen-rust
 ```
 
+To run just `codegen` tests for a single language (replace rust with language of choice: `go`, `c`, `csharp`, etc.) and a single wit file (replace `flags` with whatever wit file should be tested):
```
Comment (silesmo) [inline tests/README.md:28]: Duplicated in the text below?

Comment (yowl) [reply tests/README.md:28]: It is, but this is the specific instruction for running a `codegen` test for a single language and single wit, which wasn't covered explicitly elsewhere.

File: crates/csharp/src/lib.rs (hunk 1 above)
Comment (yowl) [reply crates/csharp/src/lib.rs]: Thanks, updated to use `ushort` and `byte`.

**Pull request merged**